core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -1651,6 +1652,54 @@ func Test_CreatedAfterQueriesWithBackfill(t *testing.T) {
 	}
 }
 
+func Test_PruneOldBlocks(t *testing.T) {
+	ctx := testutils.Context(t)
+
+	tests := []struct {
+		name          string
+		blockToCreate int
+		blocksLeft    int
+		wantErr       bool
+	}{
+		{
+			name:          "returns error if no blocks yet",
+			blockToCreate: 0,
+			wantErr:       true,
+		},
+		{
+			name:          "returns if there is not enough blocks in the db",
+			blockToCreate: 10,
+			blocksLeft:    10,
+		},
+		{
+			name:          "prunes matching blocks",
+			blockToCreate: 2000,
+			blocksLeft:    1010, // last finalized block is 10 block behind
```

Comment:
Should we use `lp.keepFinalizedBlocksDepth` here instead of hard-coded 1000?  Slightly more readable, and won't break if someone updates the default